compiler_opt/rl/train_bc.py

```diff
@@ -81,11 +86,49 @@ def train_eval(agent_config_type=agent_config.BCAgentConfig,
       batch_size=batch_size,
       train_sequence_length=train_sequence_length)
 
+  def evaluation_hook():
+    # Pytype complains that tf.compat does not have an attribute v1 at this
+    # callsite for some reason.
+    # pytype: disable=module-attr
+    global_step = tf.compat.v1.train.get_or_create_global_step()
+    # pytype: enable=module-attr
+    eval_dataset = data_reader.create_tfrecord_dataset_fn(
+        agent_cfg=agent_cfg,
+        batch_size=batch_size,
+        train_sequence_length=train_sequence_length,
+        shuffle_repeat_count=1)
+    percentage_correct = tf.keras.metrics.Accuracy()
+    batch_count = 0
+    start_time = time.time()
+    for experience in eval_dataset(_DATA_PATH.value):
+      experience_time_step = trajectories.TimeStep(experience.step_type,
+                                                   experience.reward,
+                                                   experience.discount,
+                                                   experience.observation)
+      policy_actions = agent.policy.action(experience_time_step)
+      percentage_correct.update_state(experience.action, policy_actions.action)
+
+      if (batch_count + 1) % evaluation_logging_interval == 0:
+        logging.info('Evaluating: batch = %d', batch_count + 1)
+        time_since_last_log = time.time() - start_time
+        batches_per_sec = evaluation_logging_interval / time_since_last_log
+        logging.info('%.3f batches/sec', batches_per_sec)
+        start_time = time.time()
+      batch_count += 1
```

Comment:
if you bumped batch_count above the `if`, you could use `batch_count` without adding 1 again, right?